src/big/json.cr

```diff
@@ -10,3 +10,17 @@ def BigFloat.new(pull : JSON::PullParser)
   pull.read_float
   BigFloat.new(pull.raw_value)
 end
+
+def BigDecimal.new(pull : JSON::PullParser)
+  value = case pull.kind
```

Comment:
This doesn't format, please use
```cr
case pull.kind
when :int
  pull.read_int
  value = pull.raw_value
when :float
  pull.read_float
  value = pull.raw_value
else
  value = pull.read_string
end
```

Reply:
`read_int` and `read_float` might overflow.

Reply:
`read_int` and `read_float` just tell the pull parser to check for an int/float. `raw_value` will always have the raw string.

But here we can just use `read_next` so "check that it's an int" is skipped (because it's considered in the `case`).

I added `raw_value` a lot of time ago to support BigFloat/BigInt. It makes JSON parsing a bit slower, but generally more useful.

Reply:
Ok changed, hopefully good to go now.